### PR TITLE
Add doc_fragment and module util for NetApp SolidFire

### PR DESCRIPTION
**Pull request by timuster**
##### ISSUE TYPE

- New Module Pull Request
##### COMPONENT NAME

- SolidFire storage platform
##### ANSIBLE VERSION

```
ansible 2.1.1.0
  config file = 
  configured module search path = Default w/o overrides
```
##### SUMMARY

The change adds modules that enable the management of a NetApp SolidFire storage platform. The modules were based on a design/approach similar to the existing NetApp E-Series (SANtricity) storage platform.

### CODE REVIEW

**Comment by timuster** [comment]
Migrated from https://github.com/ansible/ansible-modules-extras/pull/3259

File: lib/ansible/modules/storage/netapp/sf_account_manager.py [hunk 1]
```diff
@@ -0,0 +1,284 @@
...
+        required: true
+        description:
+        - username
+
+    password:
+        required: true
+        description:
+        - password
+
+    action:
```
Comment (thaumos) [inline lib/ansible/modules/storage/netapp/sf_account_manager.py]: Parameter of **action** is against best practices for module dev.  Please use **state** in place of this.

File: lib/ansible/modules/storage/netapp/sf_account_manager.py [hunk 2]
```diff
@@ -0,0 +1,284 @@
...
+        Before updating an account, check the previous (current) attributes to currently report
+        the 'changed' property
+
+'''
+import sys
+import json
+import logging
+from traceback import format_exc
+
+from ansible.module_utils.basic import *
```
Comment (thaumos) [inline lib/ansible/modules/storage/netapp/sf_account_manager.py]: import specific functions, not *

File: lib/ansible/modules/storage/netapp/sf_account_manager.py [hunk 3]
```diff
@@ -0,0 +1,284 @@
...
+
+import socket
+logger = logging.getLogger(__name__)
+logger.setLevel(logging.DEBUG)
+
+logging.basicConfig(level=logging.ERROR)
+logger = logging.getLogger(__name__)
+logger.setLevel(logging.DEBUG)
+
+from solidfire.factory import ElementFactory
```
Comment (thaumos) [inline lib/ansible/modules/storage/netapp/sf_account_manager.py]: Import custom packages in try/except and handled with fail_json() in main() e.g.::

    try:
        import foo
        HAS_LIB=True
    except:
        HAS_LIB=False

**Review by thaumos — CHANGES_REQUESTED** [review]
Hi @timuster, 

Thanks again for this submission as well!

Same as other PR.  Remove README file, as it is not needed.  Add in module utils and doc fragments for the modules.

Make changes to suggested lines (not comprehensive).

**Comment by timuster** [comment]
Thank you for the review, @thaumos ! I'll push the suggested changes by January 16th.

**Comment by timuster** [comment]
https://github.com/ansible/ansible/pull/19607

**Pull request closed without merging**